src/sqlfluff/rules/L052.py

```diff
@@ -224,27 +218,51 @@ def _handle_semicolon_newline(
             )
         else:
             fixes.extend(
-                [
-                    LintFix.create_after(
-                        self._choose_anchor_segment(
-                            context, "create_after", anchor_segment, filter_meta=True
-                        ),
-                        [
-                            NewlineSegment(),
-                            SymbolSegment(raw=";", type="symbol", name="semicolon"),
-                        ],
-                    ),
-                    LintFix.delete(
-                        context.segment,
-                    ),
-                ]
+                self._create_semicolon_and_delete_whitespace(
+                    context,
+                    anchor_segment,
+                    info.whitespace_deletions,
+                    [
+                        NewlineSegment(),
+                        SymbolSegment(raw=";", type="symbol", name="semicolon"),
+                    ],
+                )
```

Comment:
This code was practically identical to the buggy code. I refactored it to use the same helper function, thus it gets the same fix.